src/Common/BufferAllocationPolicy.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "config.h"
+
+#include "logger_useful.h"
+
+#include <list>
+
+namespace DB
+{
+
+class IBufferAllocationPolicy
+{
+public:
+
+    struct Settings
+    {
```

Comment:
Any comments for these fields would be useful. For example, what is the difference between `max_size` and `max_single_size`?

Reply:
addressed it